compiler/lib/Dialect/Linalg/Transforms/Transforms.cpp

```diff
@@ -1298,18 +1305,21 @@ mlir::scf::tileConsumerAndFuseProducerUsingSCFForOpExt(
         auto result = unfusedOp->getResult(i);
 
         auto effectiveUseCnt =
-            llvm::count_if(result.getUses(), [](OpOperand &opOperand) {
+            llvm::count_if(result.getUses(), [&](OpOperand &opOperand) {
+              if (unfusedOpsSet.contains(opOperand.getOwner()))
+                return false;
+
               if (auto dstOp = dyn_cast<DestinationStyleOpInterface>(
                       opOperand.getOwner())) {
                 return !dstOp.isDpsInit(&opOperand);
               }
+
               return !isa<tensor::DimOp>(opOperand.getOwner());
             });
 
-        bool hasOneOrZeroUseGeneral =
-            unfusedOp == consumer ? effectiveUseCnt < 1 : effectiveUseCnt <= 1;
-
-        bool hasOneOrZeroUseForExtract = effectiveUseCnt <= 1;
+        bool hasZeroUse = effectiveUseCnt == 0;
```

Comment:
not sure about this logic. Can we check this clean-up  with flash-attention?

Reply:
The tiling-and-fusion result of flash-attention is right. I also made the flash-attention test more restricted.